Replace ops with compiles decorator by regular registry

datapythonista opened the pull request:
Having operations defined in two places make things more complex for no reason. Removing the `compiles` decorator for most backends, and using the regular registry instead.

`pyspark` seems to have mostly all operations defined using `compiles`. I think it'd make sense to have a registry like the other backends. And that would help if we move towards #2659. But will open an issue for that, and address it separately.

datapythonista commented:
This is blocked by #2727. Once the SQLAlchemy backend code is refactored, we need to move the ops there to its `registry.py` file, and remove the `compiles` decorator.

Also, there are couple of examples in the tutorial using the `compiles` decorator that need to be updated.

jreback commented:
can you merge master and let the ci run again here

jreback commented on ibis/backends/base/sql/alchemy/translator.py:
```diff
@@ -82,8 +77,18 @@ def _nullifzero(expr):
 def _true_divide(t, expr):
+    # Having imports here, so this is self-contained and faster to clean up
+    # when this function is moved to the registry
+    import ibis.expr.types as ir
```
this doesn't make sense, these are standard ibis imports

jreback reviewed (changes requested):
rebase as well

datapythonista commented:
Rebased, still all green. Thanks @jreback 

I also moved the imports to the top. Agree that they belong here, but should be able to delete that function soon, so left them there to remove it quicker with all its imports. Not a big deal.

jreback commented:
> Rebased, still all green. Thanks @jreback
> 
> I also moved the imports to the top. Agree that they belong here, but should be able to delete that function soon, so left them there to remove it quicker with all its imports. Not a big deal.

thanks, yeah certainly removing that function is fine (followup)
